pkg/api/tag_test.go

```diff
@@ -276,7 +278,7 @@ func TestTags(t *testing.T) {
 		if err != nil {
 			t.Fatal(err)
 		}
-		tagValueTest(t, uint32(tagId), 3, 3, 0, 0, 0, 3, expectedHash, client)
+		tagValueTest(t, uint32(tagId), 4, 4, 0, 0, 0, 4, expectedHash, client)
```

Comment:
why are there 4 chunks here?

Reply:
This is according to the new behavior. For each file, we will create a new manifest. The manifest will be similar to a directory with a single file. There will be rootManifestNode entry which will have the index file metadata pointing to the file we are uploading. This will ensure we can serve on ```/bzz/{reference}``` endpoint. There will be a separate file entry with the file metadata. Due to this structure we will now have additional references.

You can check the "traversal_test.go" file. I have added comments about the extra references. For eg, file with single chunk will be stored as follows:
```
"ae16fb27474b41273c0deb355e4405d3cd0a6639f834285f97c75636c9e29df7", // root manifest                                                                                                                    
"0cc878d32c96126d47f63fbe391114ee1438cd521146fc975dea1546d302b6c0", // mainifest root metadata                                                                                                          
"05e34f11a0967e8c09968b69c4f486f569ef58a31a197992e01304a1e59f8e75", // manifest file entry                                                                                                              
"e94a5aadf259f008b7d5039420c65d692901846523f503d97d24e2f077786d9a", // bytes 
```

If user doesn't specify a file name, we use the file reference as the name. The manifest doesn't support empty paths. So we use file reference string as default name. As this is larger than what a single value in the manifest can hold, we actually add some edges. This makes the hashes count at 6. For eg:

```
"7e0a4b6cd542eb501f372438cbbbcd8a82c444740f00bdd54f4981f487bcf8b7", // root manifest                                                                                                                    
"0cc878d32c96126d47f63fbe391114ee1438cd521146fc975dea1546d302b6c0", // manifest root metadata                                                                                                           
"3f538c3b5225111a79b3b1dbb5e269ca2115f2a7caf0e6925b773457cdef7be5", // manifest file entry (Edge)                                                                                                       
"2f09e41846a24201758db3535dc6c42d738180c8874d4d40d4f2924d0091521f", // manifest file entry (Edge)                                                                                                       
"b2662d17d51ce734695d993b44c0e2df34c3f50d5889e5bc3b8718838658e6b0", // manifest file entry (Value)                                                                                                      
"f833c17be12d68aec95eca7f9d993f7d7aaa7a9c282eb2c3d79ab26a5aeaf384", // bytes        
```